public/app/features/query/components/QueryEditorRow.tsx

```diff
@@ -240,17 +246,11 @@ export class QueryEditorRow<TQuery extends DataQuery> extends PureComponent<Prop
     }
   }
 
-  isWaitingForDatasourceToLoad(): boolean {
-    // if we not yet have loaded the datasource in state the
-    // ds in props and the ds in state will have different values.
-    return this.props.dataSource.uid !== this.state.loadedDataSourceIdentifier;
-  }
-
   renderPluginEditor = () => {
     const { query, onChange, queries, onRunQuery, onAddQuery, app = CoreApp.PanelEditor, history } = this.props;
     const { datasource, data } = this.state;
 
-    if (this.isWaitingForDatasourceToLoad()) {
+    if (this.state.isLoadingDataSource) {
```

Comment:
this will break some scenarios

Reply:
There is a render before the state change, so this will be false, and the previous editor will be rendered once before the loading is initiated